### PR TITLE
Bump coredns to 1.16.4

### PR DESCRIPTION
**Pull request by olljanat**
Provider coredns extra config support from https://github.com/coredns/helm/pull/44 and solves my feature request on https://github.com/rancher/rke2/issues/1808

### CODE REVIEW

File: packages/rke2-coredns/package.yaml
Comment (brandond) [inline packages/rke2-coredns/package.yaml:2]: You can drop this back down to 0 when bumping the upstream chart version; the package version only needs to be incremented if changing the local customizations without bumping the upstream chart version.
```suggestion
packageVersion: 00
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: packages/rke2-coredns/package.yaml
Comment (olljanat) [reply packages/rke2-coredns/package.yaml]: Ok. Fixed

**Review by brandond — APPROVED** [review]

**Comment by olljanat** [comment]
ping @manuelbuil I see that you have you have reviewed coredns related changes earlier.

**Review by manuelbuil — APPROVED** [review]
LGTM! After this is merged, we need to update rke2 to use it

**Pull request merged**
